app/src/main/res/layout/fragment_login.xml

```diff
@@ -0,0 +1,110 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".ui.example.login.LoginFragment">
+
+    <android.support.constraint.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <ImageView
+            android:id="@+id/vLoginBannerImageView"
+            android:layout_width="match_parent"
+            android:layout_height="193dp"
```

Comment:
We use dimens.xml file to save all dimen values. Try to solve this harcoded value

Reply:
There something that I really don't like about this. If we leave image view like that, it could break view in devices with smaller screen. I would suggest to read a little bit from android:weight attribute

Reply:
✔ **Done**

Question: Sometimes when I put the **0dp** value from dimens.xml file, android studio suggest me to use the hardcoded value in order to optimize some things. I used the dimens.xml value but, what I have to do in this case?